src/core/server/plugins/plugins_service.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { filter, first, tap, toArray } from 'rxjs/operators';
+import { CoreService } from '../../types/core_service';
+import { ConfigService, Env } from '../config';
+import { Logger, LoggerFactory } from '../logging';
+import { PluginDiscoveryErrorType } from './plugin_discovery_error';
+import { PluginsConfig } from './plugins_config';
+import { PluginsDiscovery } from './plugins_discovery';
+
+export class PluginsService implements CoreService {
+  private readonly log: Logger;
+  private readonly discovery: PluginsDiscovery;
+
+  constructor(
+    readonly env: Env,
+    readonly logger: LoggerFactory,
+    private readonly configService: ConfigService
+  ) {
+    this.log = logger.get('plugins', 'service');
+    this.discovery = new PluginsDiscovery(logger.get('plugins', 'discovery'));
+  }
+
+  public async start() {
+    this.log.debug('starting plugins service');
+
+    const { plugins$, errors$ } = this.discovery.discover(
+      this.configService.atPath('plugins', PluginsConfig).pipe(first())
+    );
+
+    await plugins$
```

Comment:
__note:__ these two `await`'s just so that we actually trigger plugin discovery and make sure it's not broken accidentally by other changes, my plan is to have this in master, but I can get rid of them if you think it's too heavy/risky to have it in master until plugin service is fully functional.